### PR TITLE
ETSWORK-10675 - Initial Electron support with authentication to Workgrid

### PR DESCRIPTION
**Pull request by coreyjv**
With this PR developers can run the Electron app locally and authenticate to Workgrid. Other work such as Workgrid icons, and CI/CD will come in subsequent PRs.

### CODE REVIEW

**Comment by Kikketer** [comment]
Is it too late to request changing the `/electron` folder to something like `/desktop` or the like?  Naming it after the technology may not be entirely obvious what it is or what it does.

**Comment by coreyjv** [comment]
> Is it too late to request changing the `/electron` folder to something like `/desktop` or the like? Naming it after the technology may not be entirely obvious what it is or what it does.

I would say this is unsupported. The `@capacitor-community/electron` package sets this and follows the same convention as its parent project with naming the folder after the platform (e.g. `ios` and `android`).

**Comment by Kikketer** [comment]
> > Is it too late to request changing the `/electron` folder to something like `/desktop` or the like? Naming it after the technology may not be entirely obvious what it is or what it does.
> 
> I would say this is unsupported. The `@capacitor-community/electron` package sets this and follows the same convention as its parent project with naming the folder after the platform (e.g. `ios` and `android`).

yeah it's fine if we can't change it, better to not try to fight the setup that's given to us.  Technically iOS and Android aren't the "tech" they are the "platform".  So "desktop" is platform but "electron" is tech.
iOS = Swift + WebView
Android = Kotlin + WebView
Desktop = Electron... Renderer? Main? :)

Splitting hairs.  If it works, it's fine and we don't need to make it more complicated than it needs to be for a simple word/named folder.

**Review by Kikketer — DISMISSED** [review]

File: electron/capacitor.config.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "appId": "com.workgrid.app",
+  "appName": "Workgrid",
+  "bundledWebRuntime": false,
+  "npmClient": "npm",
```
Comment (TuckerWhitehouse) [inline electron/capacitor.config.json:5]: I like how this suggests you could use a different one, which you absolutely can't...

File: src/services/authentication-service.ts [hunk 2]
```diff
@@ -33,6 +33,14 @@ export function authenticationServiceFactory(
   const platform = Capacitor.getPlatform()
 
   switch (platform) {
+    case 'electron':
+      return new CognitoAuthenticationService({
+        ...config,
+        platform: 'web',
+        redirectUri: `capacitor-electron://-/login`,
+        logoutUrl: `capacitor-electron://-/login`,
```
Comment (TuckerWhitehouse) [inline src/services/authentication-service.ts:41]: How would this work if someone had multiple capacitor apps installed?

**Review by TuckerWhitehouse — DISMISSED** [review]

File: src/services/authentication-service.ts (hunk 2 above)
Comment (coreyjv) [reply src/services/authentication-service.ts:41]: I suspect someone else could authenticate if they created their own app. I think it's analogous to someone creating an installing an app on their mobile device with the same bundle identifier. In reality there isn't much we can do. The maintainer of `@capacitor-community/electron` will be releasing the ability to customize this scheme shortly per the discussion him and I have had [here](https://github.com/capacitor-community/electron/issues/71#issuecomment-842601041). That will allow us to at least change from the default.

File: electron/capacitor.config.json (hunk 1 above)
Comment (coreyjv) [reply electron/capacitor.config.json:5]: The funny part is that when I converted the capacitor config to TypeScript it complained that `npmClient` wasn't an allowed option :)

Comment (TuckerWhitehouse) [reply electron/capacitor.config.json:5]: hahah

File: src/services/authentication-service.ts (hunk 2 above)
Comment (coreyjv) [reply src/services/authentication-service.ts:41]: @TuckerWhitehouse -- The capability to customize the scheme was added in `3.0.0-rc.5`. I've updated this PR to reflect that using the `com.workgrid.client` to be consistent with our other clients.

**Review by TuckerWhitehouse — APPROVED** [review]

**Review by Kikketer — APPROVED** [review]

**Pull request merged**